### PR TITLE
(GH-27) fix creation path of 'item' templates

### PR DESCRIPTION
**Pull request by da-ar**
This commit fixes an issue whereby `item` templates would be added to the parent directory rather than the current working directory (CWD).

By default `pct new` will add content from templates in the CWD.  The behaviour can be overridden by supplying the `--output` flag

### CODE REVIEW

**Review by sanfrancrisko — APPROVED** [review]
LGTM - happy to merge, but, wondering if tests are easy to add here? I seem to recall you did mention it wouldn't be in the current state?

**Comment by da-ar** [comment]
> LGTM - happy to merge, but, wondering if tests are easy to add here? I seem to recall you did mention it wouldn't be in the current state?

There are tests and they work just fine right now. I should have reviewed them and ensured that there were tests for the conditions we found. Hold off on the merge. ty.

File: internal/pkg/pct/pct.go
Comment (sanfrancrisko) [inline internal/pkg/pct/pct.go:87]: A minor drawback of this pattern is the amount of objects with similar names created. We're going to have to have a think about this - maybe using letter identifiers or something along those lines.

For me `wrapper` is a bit confusing in the context of where it's called. Right here - OK, because I can see how it was constructed above. 

Perhaps:
```go
type osWrapper interface {}
type osFunc struct {}
var osUtils osWrapper = osFunc{}
```

**Review by sanfrancrisko — COMMENTED** [review]
LGTM - just a minor comment about object naming

File: cmd/new/new.go
Comment (jpogran) [inline cmd/new/new.go:71]: I need to add more code comments here. This line was added to account for `pct new puppet-class foo`.  There may be better ways of handling that with cobra, but we shouldn't remove the functionality.

I also need to start a UX sheet of command workflows we support.

**Review by jpogran — DISMISSED** [review]
We need to split this PR into two: one for the new unit testing approach and another for this fix. For the unit testing approach, we'll need to work through naming and api surface.

For the bug PR (this one) we need to add detail to the commit message and the PR description explaining the bug being fixed and the resolution.

**Comment by jpogran** [comment]
I updated the ticket with a better explanation of how the parameters should behave (but don't in the case of items).

**Review by jpogran — CHANGES_REQUESTED** [review]
David paired with Ciaran and myself and explained the approach for this type of testing. We'll merge this in as is and work to mature our testing story as we mature the PCT package.

We just need a commit message explaining the bug, the fix and the change in usage

File: cmd/new/new.go
Comment (da-ar) [reply cmd/new/new.go:71]: added this back in but as discussed this is more magic

File: internal/pkg/pct/pct.go
Comment (da-ar) [reply internal/pkg/pct/pct.go:87]: name changed

**Review by jpogran — APPROVED** [review]

**Pull request merged**
